File: data/src/main/java/com/github/anrimian/musicplayer/data/models/composition/InitialSource.java
package com.github.anrimian.musicplayer.data.models.composition;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef(value = {
        InitialSource.LOCAL,
        InitialSource.REMOTE
})
@Retention(RetentionPolicy.SOURCE)
@Deprecated
public @interface InitialSource {
    int LOCAL = 1;
    int REMOTE = 2;
}

File: domain/src/main/java/com/github/anrimian/musicplayer/domain/models/composition/InitialSource.kt
package com.github.anrimian.musicplayer.domain.models.composition


enum class InitialSource(val id: Int) {
    LOCAL(1),
    REMOTE(2),
    APP(3);

    companion object {
        private val idMap = HashMap<Int, InitialSource>().apply {
            InitialSource.entries.forEach { source -> put(source.id, source) }
        }

        fun fromId(id: Int): InitialSource {
            return idMap[id] ?: throw IllegalStateException()
        }
    }

}